ui/apps/dashboard/e2e/member clusters/test-utils.ts

```diff
@@ -0,0 +1,377 @@
+limitations under the License.
+*/
+
+import { Page, expect } from '@playwright/test';
+import * as k8s from '@kubernetes/client-node';
+import { parse } from 'yaml';
+import _ from 'lodash';
+import {
+    createKarmadaApiClient,
+    debugScreenshot
+} from '../test-utils';
+
+// Re-export utilities from parent test-utils for single-point import
+export {
+    setupDashboardAuthentication,
+    debugScreenshot
+} from '../test-utils';
+
+/**
+ * UI element selectors for member cluster forms
+ * Centralized to make maintenance easier if UI text changes
+ */
+export const MemberClusterSelectors = {
+    nameField: '* Name :',
+    kubeconfigField: /Editor content/,
+    taintKeyField: 'Please output the key of the point',
+    taintValueField: 'Please output the value of the point',
+} as const;
+
+/**
+ * Creates a configured Kubernetes API client for Karmada CustomObjects
+ * @returns Kubernetes CustomObjectsApi client
+ */
+function createKarmadaCustomObjectsApiClient(): k8s.CustomObjectsApi {
+    return createKarmadaApiClient(k8s.CustomObjectsApi);
+}
+
+/**
+ * Generate test Member Cluster kubeconfig YAML with timestamp
+ * Note: The Member Cluster creation page expects kubeconfig format, not Cluster resource format
+ */
+export function generateTestClusterKubeConfig() {
+    const timestamp = Date.now();
+    return `apiVersion: v1
+kind: Config
+clusters:
+- cluster:
+    insecure-skip-tls-verify: true
+    server: https://test-cluster-${timestamp}.example.com:6443
+  name: test-membercluster-${timestamp}
+contexts:
+- context:
+    cluster: test-membercluster-${timestamp}
+    user: test-user-${timestamp}
+  name: test-context-${timestamp}
+current-context: test-context-${timestamp}
+users:
+- name: test-user-${timestamp}
+  user:
+    token: test-token-${timestamp}`;
+}
+
+/**
+ * Generate test Member Cluster resource YAML with timestamp in Cluster resource format
+ * Note: This generates a Cluster resource that can be created via Karmada API
+ */
+export function generateTestMemberClusterResourceYaml() {
+    const timestamp = Date.now();
+    const clusterName = `test-membercluster-${timestamp}`;
+    return `apiVersion: cluster.karmada.io/v1alpha1
+kind: Cluster
+metadata:
+  name: ${clusterName}
+spec:
+  syncMode: Pull
+  apiEndpoint: https://test-cluster-${timestamp}.example.com:6443
+  secretRef:
+    namespace: karmada-cluster
+    name: ${clusterName}
+  insecureSkipCAVerification: true`;
+}
+
+/**
+ * Creates a Kubernetes Member Cluster using the Kubernetes JavaScript client.
+ * This is a more robust way to set up test data than UI interaction.
+ * @param yamlContent The YAML content of the member cluster.
+ * @returns A Promise that resolves when the member cluster is created.
+ */
+export async function createK8sMemberCluster(yamlContent: string): Promise<void> {
+    try {
+        const k8sApi = createKarmadaCustomObjectsApiClient();
+        const yamlObject = parse(yamlContent) as any;
```

Comment:
here we can use `const yamlObject = parse(yamlContent) as Record<string, any>;` as a tradeoff,  a better way is to generate the corresponding typescript definition based on the membercluster api resource